apax/model/gmnn.py

```diff
@@ -22,6 +25,34 @@
 log = logging.getLogger(__name__)
 
 
+def canonicalize_neighbors(neighbor):
+    if type(neighbor) in [partition.NeighborList, NeighborSpoof]:
```

Comment:
If you are not explicitly excluding subclasses, I'd suggest using `if isinstance(neighbor, (partition.NeighborList, NeighborSpoof))` instead

Reply:
This basically makes it a oneliner:
```py
return neighbor.idx if isinstance(...) else neighbor
```

Reply:
That's a great suggestion, I would have expected this to give an error but it works.